src/dbnode/generated/proto/annotation/annotation.proto

```diff
@@ -3,11 +3,17 @@ syntax = "proto3";
 package annotation;
 
 message Payload {
-    MetricType metric_type   = 1;
+    SourceFormat source_format = 3;
+
+    // Used when source_format == PROMETHEUS
+    PrometheusType prom_type = 1;
     bool handle_value_resets = 2;
+
+    // Used when source_format == GRAPHITE
+    GraphiteType graphite_type = 4;
```

Comment:
did you consider defining this as oneof since only 1 of these can be set?

Reply:
No because there is a lot of existing data already.